music21/metadata/__init__.py

```diff
@@ -96,161 +184,499 @@ class Metadata(base.Music21Object):
+        If you do in fact want to remove any existing items with this key,
+        you can use dictionary-style or attribute-style setting instead.
 
-        # a dictionary of Text elements, where keys are work id strings
-        # all are loaded with None by default
-        self._workIds = OrderedDict()
-        for abbreviation, workId in self.workIdAbbreviationDict.items():
-            # abbreviation = workIdToAbbreviation(id)
-            if workId in keywords:
-                self._workIds[workId] = Text(keywords[workId])
-            elif abbreviation in keywords:
-                self._workIds[workId] = Text(keywords[abbreviation])
-            else:
-                self._workIds[workId] = None
+        >>> md = metadata.Metadata()
+        >>> md.add('composer', 'Houcine Slaoui')
+        >>> md['composer']
+        (<music21.metadata.primitives.Contributor composer:Houcine Slaoui>,)
+
+        >>> md.add('title', [metadata.Text('الماريكان', language='ar'),
+        ...                  metadata.Text('The Americans',  language='en')])
+        >>> titles = md['title']
+        >>> titles
+        (<music21.metadata.primitives.Text الماريكان>,
+        <music21.metadata.primitives.Text The Americans>)
+        >>> titles[0].language
+        'ar'
+        >>> titles[1].language
+        'en'
+        '''
+        self._add(key, value, isCustom=False)
 
-        # search for any keywords that match attributes
-        # these are for direct Contributor access, must have defined
-        # properties
-        for attr in ['composer', 'date', 'title']:
-            if attr in keywords:
-                setattr(self, attr, keywords[attr])
+    def getCustom(self, key: str) -> t.Tuple[ValueType, ...]:
+        return self._get(key, isCustom=True)
+
+    def addCustom(self, key: str, value: t.Union[t.Any, t.Iterable[t.Any]]):
+        self._add(key, value, isCustom=True)
+
+    def setCustom(self, key: str, value: t.Union[t.Any, t.Iterable[t.Any]]):
+        self._set(key, value, isCustom=True)
+
+    def getAllNamedValues(self, skipContributors=False) -> t.List[t.Tuple[str, ValueType]]:
+        '''
+        Returns all values stored in this metadata as a list of (nsKey, value) tuples.
+        nsKeys with multiple values will appear multiple times in the list (rather
+        than appearing once, with a value that is a list of values).
+        The tuple's first element is either of the form 'namespace:name', or a
+        custom key (with no form at all).
+
+        >>> md = metadata.Metadata()
+        >>> md.add('composer', 'Jeff Bowen')
+        >>> md.add('librettist', 'Hunter Bell')
+        >>> md.add('title', 'Other World')
+        >>> md.addCustom('excerpt-start-measure', 1234)
+        >>> all = md.getAllNamedValues()
+        >>> all
+        [('marcrel:CMP', <music21.metadata.primitives.Contributor composer:Jeff Bowen>),
+        ('marcrel:LBT', <music21.metadata.primitives.Contributor librettist:Hunter Bell>),
```

Comment:
indent to show inside bracket.  (same w/ below, etc.)

Reply:
Done